OPAL/ll/src/main/scala/org/opalj/ll/fpcf/analyses/ifds/taint/NativeTaintProblem.scala

```diff
@@ -0,0 +1,58 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.ll.fpcf.analyses.ifds.taint
+
+import org.opalj.br.ObjectType
+import org.opalj.ifds.{AbstractIFDSNullFact, Callable, AbstractIFDSFact}
+import org.opalj.ll.llvm.value.Value
+
+trait NativeTaintFact extends AbstractIFDSFact
+
+object NativeTaintNullFact extends NativeTaintFact with AbstractIFDSNullFact
+
+/**
+ * A tainted variable.
+ *
+ * @param index The variable's definition site.
+ */
+case class JavaVariable(index: Int) extends NativeTaintFact
+case class NativeVariable(value: Value) extends NativeTaintFact
+
+/**
+ * A tainted array element.
+ *
+ * @param index The array's definition site.
+ * @param element The index of the tainted element in the array.
+ */
+case class JavaArrayElement(index: Int, element: Int) extends NativeTaintFact
+case class NativeArrayElement(base: Value, indices: Iterable[Long]) extends NativeTaintFact
+
+/**
+ * A tainted static field.
+ *
+ * @param classType The field's class.
+ * @param fieldName The field's name.
+ */
+case class JavaStaticField(classType: ObjectType, fieldName: String) extends NativeTaintFact
+
+/**
+ * A tainted instance field.
+ *
+ * @param index The definition site of the field's value.
+ * @param classType The field's type.
+ * @param fieldName The field's value.
+ */
+case class JavaInstanceField(index: Int, classType: ObjectType, fieldName: String) extends NativeTaintFact
+
+/**
+ * A path of method calls, originating from the analyzed method, over which a tainted variable
+ * reaches the sink.
+ *
+ * @param flow A sequence of method calls, originating from but not including this method.
+ */
+case class NativeFlowFact(flow: Seq[Callable]) extends NativeTaintFact {
+    override val hashCode: Int = {
```

Comment:
Why is this explicit hashCode necessary?

Reply:
It's the same as for the regular FlowFact, I guess that is from Mario?

Reply:
Actually, it's my own undocumented code :(
I guess it is to ensure that different (i.e. `ne`) flows have the same hashCode? I'm not sure anymore and I don't know why, if that's the case, `equals` is not implemented as well.